packages/pwa/app/components/_app-config/index.jsx

```diff
@@ -84,6 +72,34 @@ AppConfig.extraGetPropsArgs = (locals = {}) => {
     }
 }
 
+AppConfig.getUserPreferredLocales = (locals) => {
+    // TODO: `locals` as an argument doesn't seem right think a little bit more about this one.
+    // Like do we need the request object here?
+    const {originalUrl} = locals
+
+    const localeUrl = originalUrl && originalUrl.split('/')[1]
+
+    const locale = originalUrl
+        ? SUPPORTED_LOCALES.includes(localeUrl)
+            ? localeUrl
+            : DEFAULT_LOCALE
+        : window?.__PRELOADED_STATE__?.appProps?.targetLocale
```

Comment:
Minor, but important subtlety: `getUserPreferredLocales` should _not_ know about the app's supported locales nor its default locale. We simply find out which locales the user wants. Then later, `whichLocaleToLoad` in locale.js will match the user locales against the app locales.

```suggestion
    const locale = originalUrl
        ? localeUrl
        : window?.__PRELOADED_STATE__?.appProps?.targetLocale
```

Reply:
This has also been addressed. Thanks